tests/integration/test_fim/test_files/test_audit_buffer/test_audit_buffer_over_time.py

```diff
@@ -0,0 +1,323 @@
+            type: fixture
+            brief: Create a given list of folders when the module starts. Delete the folders at the end of the module.
+        - configure_local_internal_options_function:
+            type: fixture
+            brief: Set local_internal_options.conf file.
+        - restart_syscheck_function:
+            type: fixture
+            brief: restart syscheckd daemon, and truncate the ossec.log.
+        - wait_syscheck_start:
+            type: fixture
+            brief: check that the starting FIM scan is detected.
+
+    assertions:
+        - Verify whadata queue is full
+        - Verify all inserted files are detected in whodata mode if files are inserted after queue space is freed
+
+    input_description: The file 'configuration_audit_buffer_over_time.yaml' provides the configuration
+                       template.
+                       The file 'cases_audit_buffer_over_time_no_overflow.yaml' provides the tes cases configuration
+                       details for each test case.
+
+    expected_output:
+        - r".*(Internal audit queue is full). Some events may be lost. Next scheduled scan will recover lost data."
+        - r".*Sending FIM event: (.+)$"
+    '''
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+    whodata_events = metadata['files_first_insert'] + metadata['files_second_insert']
+
+    # Wait for FIM to process all initial whodata messages
+    time.sleep(2)
+
+    # Insert an amount of files
+    for file in range(0, metadata['files_first_insert']):
+        create_file(REGULAR, test_folders[0], f'test_file_{file}', content='')
+
+    # Wait for files to be processed
+    time.sleep(metadata['wait_time'])
+
+    # Insert a second amount of files
+    for file in range(0, metadata['files_second_insert']):
+        create_file(REGULAR, test_folders[0], f'test_file_second_insert_{file}', content='')
+
+    # Detect audit queue is full
+    with pytest.raises(TimeoutError):
+        detect_audit_queue_full(wazuh_log_monitor, update_position=False)
+
+    # Get all file events
+    results = wazuh_log_monitor.start(timeout=T_60, callback=callback_detect_file_added_event,
+                                      accum_results=whodata_events,
+                                      error_message=f"Did not receive the expected amount of \
+                                                      whodata file added events").result()
+    # Validate all files where found in whodata mode - no files where dropped
+    for result in results:
+        assert result['data']['mode'] == 'whodata', f"Expected whodata event, found {result['data']['mode']} event"
+
+
+@pytest.mark.parametrize('test_folders', [test_folders], ids='')
+@pytest.mark.parametrize('configuration, metadata', zip(t2_configurations, t2_configuration_metadata),
+                         ids=t2_test_case_ids)
+def test_audit_buffer_overflown(configuration, metadata, test_folders, set_wazuh_configuration,
```

Comment:
Typo in test name.

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/4227/commits/818122a07a426d23dde0fb26954da571040cd55b